src/docs/arch-cursor.dox

```diff
@@ -371,6 +371,28 @@ how many references to closed data handles are freed, we may continue our walk.
 usually strike a good balance between not having a lot of overhead for sweeps, and keeping up with
 the need to free up shared resources.
 
+@section arch_cursor_bound Range bounded cursors
+
+The implementation of bounds brings performance optimizations. Bounds checking detects when a key is out of range and returns WT_NOTFOUND
+regardless of whether the value associated with the key is visible to the cursor. This skips the traversal of unnecessary records.
+
+The lower and upper bound in a cursor is represented as a \c WT_ITEM. The implementation applies bounds through the function
+__wt_cursor_bound which takes the cursor and a config string as arguments. The config string specifies whether the operation is clearing
+or setting the bound, and if a "set" is in progress, whether the bound is a lower or upper bound. There are four flags that can get set
+when applying bounds: WT_CURSTD_BOUND_LOWER, WT_CURSTD_BOUND_LOWER_INCLUSIVE, WT_CURSTD_BOUND_UPPER, and WT_CURSTD_BOUND_UPPER_INCLUSIVE.
+When calling __wt_cursor_bound with valid bounds, the respective bound flag gets set and the bound key data saved in the cursor.
+Clearing a cursor's bounds can be done via resetting the cursor or calling bound with the "clear" config which will free both bounds buffers.
+
+Positioning the cursor for traversal with bounds has a separate implementation to place the cursor at the "start" or "end" of the range it can operate
```

Comment:
Reword first sentence: When calling \c cursor->next and \c cursor->prev on an unpositioned cursor WiredTiger guarantees that the cursor will start walking from the start and end of the file respectively. The range bounded cursor logic continues to uphold this guarantee. Doing so requires additional logic to place the cursor at the start or end of the range.

Reply:
[f382c49](https://github.com/wiredtiger/wiredtiger/pull/8367/commits/f382c49e92533aafae89dae772c9766b5fbc97ce)